### DISCUSSION TITLE
Form Field

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#311] [Component suggestions]
<img width="480" alt="visuele weergave van de form field component" src="https://github.com/user-attachments/assets/a013a10b-1810-41a8-b02b-0dcb123851db">

We willen graag meer leren over de Form Field component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1727676060731779)

---

## Naam

Form Field

## Doel

Een invoerveld en bijbehorende onderdelen die helpen bij het invullen.

## Gerelateerde componenten

[Checkbox Group](https://github.com/orgs/nl-design-system/discussions/223), [Date Input](https://github.com/orgs/nl-design-system/discussions/270), [File Input](https://github.com/orgs/nl-design-system/discussions/310), [Number Input](https://github.com/orgs/nl-design-system/discussions/235), [Password Input](https://github.com/orgs/nl-design-system/discussions/285), [Radio Group](https://github.com/orgs/nl-design-system/discussions/222), [Range](https://github.com/orgs/nl-design-system/discussions/231), [Text Area](https://github.com/orgs/nl-design-system/discussions/297), [Text Input](https://github.com/orgs/nl-design-system/discussions/225), [Text Input Group](https://github.com/orgs/nl-design-system/discussions/296)

## Onderzoek

[Always use a label - Adam Silver](https://adamsilver.io/blog/always-use-a-label/)
Adam Silver legt uit waarom labels altijd toegepast moeten worden en zichtbaar moeten zijn.

[4 common mistakes UI/UX designers make when trying to help users spot required form fields (and what user research shows is better) - Adam Silver](https://adamsilver.io/blog/how-to-highlight-required-and-optional-form-fields/)
Adam Silver legt uit hoe je verplichte en optionele velden het beste kunt markeren.

[Website Forms Usability: Top 10 Recommendations - Nielsen Norman Group](https://www.nngroup.com/articles/web-form-design/)
Checklist voor ontwerp van formulieren waarbij enkele punten betrekking hebben tot Form Field Labels.

[Avoid Messages Under Fields - Adrian Roselli](http://adrianroselli.com/2017/01/avoid-messages-under-fields.html)
Artikel dat beschrijft waarom Form Field Descriptions beter tussen het label en form control geplaatst kunnen worden.

[The problem with placeholders and what to do instead - Adam Silver](https://adamsilver.io/blog/the-problem-with-placeholders-and-what-to-do-instead/)
Artikel dat beschrijft waarom placeholders voor problemen zorgen.

[Avoid Messages Under Fields - Adrian Roselli](http://adrianroselli.com/2017/01/avoid-messages-under-fields.html)
Artikel dat beschrijft waarom Form Field Error Messages beter tussen het label en form control geplaatst kunnen worden.

[How to make inline error messages accessible - Hidde de Vries](https://hidde.blog/how-to-make-inline-error-messages-accessible/)
Artikel waarin Hidde beschrijft hoe je inline error messages toegankelijk kunt aanbieden.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-field--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=21379-6977&node-type=canvas&t=EqutN9S1iqtwSNR4-11)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/field)

---

### Screenshots / Varianten

**Storybook**

<img width="800" alt="image" src="https://github.com/user-attachments/assets/c65a0252-d2c1-4eec-8169-462e30678e79"><br>

With description
A Field can have a description.
<img width="800" alt="image" src="https://github.com/user-attachments/assets/fecc8fde-2919-4915-a9a0-dd30a2a0a648">

With error
A Field can indicate if the contained input has a validation error. Use Error Message to describe the error. 
<img width="800" alt="image" src="https://github.com/user-attachments/assets/f5094ae1-6880-4deb-ab18-40a2368a5b2f">

**Figma**

Optional fields
<img width="400" alt="image" src="https://github.com/user-attachments/assets/c7731dd6-698f-45d0-a804-3f511989578d">

With description
<img width="400" alt="image" src="https://github.com/user-attachments/assets/3a81fbcd-34ff-4578-b63f-40d10637bf91">

With character count
<img width="400" alt="image" src="https://github.com/user-attachments/assets/ee916cd3-4948-4862-8a1d-405052442173">

Invalid
<img width="400" alt="image" src="https://github.com/user-attachments/assets/272af04b-0481-460f-8a52-7a0eec018648">

---

### Notities
- Onder de naam: Field.
- Er is een maximale breedte ingesteld voor de Field.
- Form field label, Form field description en Form field error message zijn erin meegenomen.
- In Figma is ook een Character count te zien onder de Text input of Select.
- Richtlijnen beschikbaar.

**Reply by VincentSmedinga** [reply]
We hebben ervoor gekozen om Field genoeg te laten zijn als naam, dus zonder Form ervoor. We verwachten het concept ‘Field’ niet buiten formulieren nodig te hebben of accepteren dat we in dat geval een andere naam moeten vinden.

Als je hier wel Form ervoor zet moet je ook Form Field Set aanhouden (in plaats van Field Set, terwijl het HTML-element gewoon fieldset is) en Form Field Description, Form Field Error Message, Form Field Character Counter etc. en dan wordt het wel wat lang allemaal.

Zie ook [mijn reactie bij Fieldset](https://github.com/orgs/nl-design-system/discussions/309#discussioncomment-10914564).

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/react-input-form-field--docs)
- [Figma - WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=958-1925&node-type=canvas&t=86afwoesAh1eWIPu-11)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/FormField)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="560" alt="image" src="https://github.com/user-attachments/assets/fb4c6da1-9f1c-4b55-8432-79b2dccef96e">

With description
<img width="560" alt="image" src="https://github.com/user-attachments/assets/3169ce4d-c95a-4827-991b-6c19a4054366">

Invalid
<img width="560" alt="image" src="https://github.com/user-attachments/assets/1c30c2f2-41f1-4bf6-abfa-af71d20026b0">

Single checkbox
<img width="560" alt="image" src="https://github.com/user-attachments/assets/60b672fe-0bd7-4643-a0eb-ef93ca76bd1b">

Checkbox group
<img width="560" alt="image" src="https://github.com/user-attachments/assets/5b871ce6-0d90-4a02-8016-1cf9c01ecab0">

Checkbox group with description
<img width="560" alt="image" src="https://github.com/user-attachments/assets/0e7d421a-520f-4e71-8248-d1a2cf122e65">

Radio button group
<img width="560" alt="image" src="https://github.com/user-attachments/assets/ed1edb83-9fe0-4518-bb40-c1b110adb0e2">

Radio button group with description
<img width="560" alt="image" src="https://github.com/user-attachments/assets/49ee6eaf-ec10-4e55-8a2e-7ac901f5666b">

Select input
<img width="560" alt="image" src="https://github.com/user-attachments/assets/12648740-764d-4a40-b839-0da2e8a85044">

**Figma**

Text input
<img width="360" alt="image" src="https://github.com/user-attachments/assets/25649fc4-4cc2-47fe-b6c5-dd7af10f2aef">

Textarea
<img width="360" alt="image" src="https://github.com/user-attachments/assets/a206444f-8f73-42f5-afa6-97d0e69c780c">

Checkbox group
<img width="360" alt="image" src="https://github.com/user-attachments/assets/a330c569-1ee3-43e8-bc93-acd27af0501e">

Radio group
<img width="360" alt="image" src="https://github.com/user-attachments/assets/f094f44f-e989-49dc-ac30-0c95ca94652b">

Description
Form field kan aanvullende informatie bevatten dankzij Form field description.
<img width="360" alt="image" src="https://github.com/user-attachments/assets/b1d6078b-5e92-4041-8c11-dd18073a4939">

Form field als Heading
Standaard bevat Form field een Form field label. Via het component panel is deze in te wisselen voor een Heading. 
<img width="360" alt="image" src="https://github.com/user-attachments/assets/25da7218-00fc-44a1-8bc8-9563c0fca697">

State default
<img width="320" alt="image" src="https://github.com/user-attachments/assets/17b25e79-8ad0-4c07-9af3-8316e151b9a0">

State invalid 
<img width="320" alt="image" src="https://github.com/user-attachments/assets/c1a4722a-3731-4147-b981-e6b0e8394475">

---

### Notities
- Verschillende inputs zijn hierin meegenomen, zoals Checkbox Group, Radio Group en Select input.
- Form field label, Form field description en Form field error message zijn hier ook meegenomen.
- Form field error message is in Storybook onder de input en in Figma boven de input. 
- In Storybook is ook een single checkbox te zien. Deze bevat geen Form field label.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/62.8.4/components/form/form-group/input)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-input--input)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/form/form-groups/input)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="560" alt="image" src="https://github.com/user-attachments/assets/fce54790-01ee-4f5d-949e-328e2ce25773">

Required
<img width="560" alt="image" src="https://github.com/user-attachments/assets/f7a782da-9cb5-472f-9898-c45d680e0096">

Disabled
<img width="560" alt="image" src="https://github.com/user-attachments/assets/1394d1d2-701a-423d-abce-65b0964c5f88">

Read-only
<img width="560" alt="image" src="https://github.com/user-attachments/assets/e40ea0d7-cd55-4fd5-a305-21e0a5410569">

Info active
<img width="560" alt="image" src="https://github.com/user-attachments/assets/f2aa0eb8-2d79-459a-bf78-9baa3906a6ee">

Info fixed false
<img width="560" alt="image" src="https://github.com/user-attachments/assets/530ef8a8-be11-4aea-9ba5-55877e476f10">

Invalid
<img width="560" alt="image" src="https://github.com/user-attachments/assets/20052613-1ee3-4285-a91e-a5efe978a876">

Feedback icon
<img width="560" alt="image" src="https://github.com/user-attachments/assets/40ef6be9-c729-4312-8766-3d8e68d133ba">

---

### Notities
- Onder de naam: Input.
- Verschillende types zijn hierin meegenomen, zoals text, email, date, password, url en tel. 
- Form field label is erin meegenomen.
- Mogelijkheid voor een icon right in de Input.
- Mogelijkheid voor een toggletip. Bij on click verschijnt er een info tekst onder het Form field label.
- Formaat (size) van Input is instelbaar.

**Comment by Rozerinay** [comment]
## RVO

[GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-field)

---

### Notities
Heeft alleen een Form field component in GitHub.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-form-field--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=958-1925&node-type=canvas&t=uusUWElC3vWT3MCJ-11)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/form-field)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="480" alt="image" src="https://github.com/user-attachments/assets/7669ff85-c34d-42ed-a4a6-bd14fcfb9436">

Invalid
<img width="480" alt="image" src="https://github.com/user-attachments/assets/0c4135de-c46c-4d0c-bbb9-55c826a94bbc">

**Figma**

Form field kan verschillende form inputs bevatten. Standaard bevat Form field een Text input.

<img width="344" alt="image" src="https://github.com/user-attachments/assets/b21ae325-42cb-4f72-a1e2-aaf460c794af">

Default
<img width="500" alt="image" src="https://github.com/user-attachments/assets/4558ecbe-f1d8-47b3-9a21-92b001efd224">

Invalid
<img width="500" alt="image" src="https://github.com/user-attachments/assets/10e7c24c-4969-49e7-ae98-bc53f1383503">

Text input
<img width="350" alt="image" src="https://github.com/user-attachments/assets/f1977439-4832-4838-bd03-81606e9152e4">

Text area
<img width="350" alt="image" src="https://github.com/user-attachments/assets/93906907-2314-4aa2-ba77-1b7bcb61e149">

Checkbox group
<img width="350" alt="image" src="https://github.com/user-attachments/assets/584cabed-70d0-4cb5-97ff-03f0a0e34ab4">

Radio group
<img width="350" alt="image" src="https://github.com/user-attachments/assets/bbdfde11-d053-44d4-99bc-20847e5401e0">

Description
<img width="350" alt="image" src="https://github.com/user-attachments/assets/bba6033a-72c8-435d-97c4-99e22060d40f">

Form field label als Heading
<img width="350" alt="image" src="https://github.com/user-attachments/assets/50d4a8d6-767b-466e-9a7c-e8a6082ff1f5">


---

### Notities
- In Figma zijn verschillende inputs hierin meegenomen, zoals Text Input, Textarea, Checkbox Group en Radio Group.
- Form field label, Form field description en Form field error message zijn erin meegenomen.
- Form field error message is in Storybook onder de input en in Figma boven de input.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Form Field op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=form+field+176)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Form Field op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=form+field+176)
